### PR TITLE
makes all virus chems deal toxin damage when ingested

### PR DESCRIPTION
**Pull request by boodaliboo**
your drinking bacteria plasma mixed with bacteria mutagen mixed with bacteria and uranium infused liquids, you will feel sick
#### Changelog

:cl:  
tweak: makes virus chems deal toxin damage
/:cl:

### CODE REVIEW

**Comment by boodaliboo** [comment]
fuck

**Comment by boodaliboo** [comment]
why did you put thumbs down xantam is it because i thumbs downed yours?

**Comment by Xantam-13** [comment]
Don't like this. There are not enough doctors in game who actually know what they are doing, enough to treat serious toxin damage from virologists or the like.

**Comment by boodaliboo** [comment]
> Don't like this. There are not enough doctors in game who actually know what they are doing, enough to treat serious toxin damage from virologists or the like.

this does incredibly little toxic damage and these work exactly like normal chem just drink some charcoal

**Comment by MegaEmpirical** [comment]
They'll manage or die trying. MERGE.

**Comment by boodaliboo** [comment]
still having trouble figuring out the errors help is appreciated

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -1509,45 +1509,101 @@
...
 	color = "#A3C00F" // rgb: 163,192,15
+	var/toxpwr = 1.5
 	taste_description = "sourness"
 
+/datum/reagent/toxin/mutagen/mutagenvirusfood/on_mob_life(mob/living/carbon/M)
+	if(toxpwr)
+		M.adjustToxLoss(toxpwr*REM, 0)
+		. = TRUE
+	..()
+
```
Comment (Hopekz) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	M.adjustToxLoss(toxpwr*REM, 0)
	..()

```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 2]
```diff
@@ -1509,45 +1509,101 @@
...
 	name = "sucrose agar"
 	color = "#41B0C0" // rgb: 65,176,192
+	var/toxpwr = 0.5
 	taste_description = "sweetness"
+	
+/datum/reagent/toxin/mutagen/mutagenvirusfood/sugar/on_mob_life(mob/living/carbon/M)
+	if(toxpwr)
+		M.adjustToxLoss(toxpwr*REM, 0)
+		. = TRUE
+	..()
```
Comment (Hopekz) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	M.adjustToxLoss(toxpwr*REM, 0)
	..()

```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 3]
```diff
@@ -1509,45 +1509,101 @@
...
 	name = "virus rations"
 	color = "#D18AA5" // rgb: 209,138,165
+	var/toxpwr = 0.25
 	taste_description = "bitterness"
+	
+/datum/reagent/medicine/synaptizine/synaptizinevirusfood/on_mob_life(mob/living/carbon/M)
+	if(toxpwr)
+		M.adjustToxLoss(toxpwr*REM, 0)
+		. = TRUE
+	..()
```
Comment (Hopekz) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	M.adjustToxLoss(toxpwr*REM, 0)
	..()

```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 4]
```diff
@@ -1509,45 +1509,101 @@
...
 	color = "#A69DA9" // rgb: 166,157,169
+	var/toxpwr = 1
 	taste_description = "bitterness"
 	taste_mult = 1.5
+	
+/datum/reagent/toxin/plasma/plasmavirusfood/on_mob_life(mob/living/carbon/M)
+	if(toxpwr)
+		M.adjustToxLoss(toxpwr*REM, 0)
+		. = TRUE
+	..()
```
Comment (Hopekz) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	M.adjustToxLoss(toxpwr*REM, 0)
	..()

```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 5]
```diff
@@ -1509,45 +1509,101 @@
...
+	var/toxpwr = 0.5
 	taste_description = "bitterness"
 	taste_mult = 1.5
 
+/datum/reagent/toxin/plasma/plasmavirusfood/weak/on_mob_life(mob/living/carbon/M)
+	if(toxpwr)
+		M.adjustToxLoss(toxpwr*REM, 0)
+		. = TRUE
+	..()
+
```
Comment (Hopekz) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	M.adjustToxLoss(toxpwr*REM, 0)
	..()

```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 6]
```diff
@@ -1509,45 +1509,101 @@
...
 	color = "#67ADBA" // rgb: 103,173,186
+	var/toxpwr = 2.5
 	taste_description = "the inside of a reactor"
 
+/datum/reagent/uranium/uraniumvirusfood/on_mob_life(mob/living/carbon/M)
+	if(toxpwr)
+		M.adjustToxLoss(toxpwr*REM, 0)
+		. = TRUE
+	..()
+
```
Comment (Hopekz) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	M.adjustToxLoss(toxpwr*REM, 0)
	..()

```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 7]
```diff
@@ -1509,45 +1509,101 @@
...
+	var/toxpwr = 2
 	color = "#2FF2CB" // rgb: 47,242,203
 	taste_description = "the inside of a reactor"
 
+/datum/reagent/uranium/uraniumvirusfood/unstable/on_mob_life(mob/living/carbon/M)
+	if(toxpwr)
+		M.adjustToxLoss(toxpwr*REM, 0)
+		. = TRUE
+	..()
+
```
Comment (Hopekz) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	M.adjustToxLoss(toxpwr*REM, 0)
	..()

```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 8]
```diff
@@ -1509,45 +1509,101 @@
...
 	color = "#04506C" // rgb: 4,80,108
+	var/toxpwr = 1.5
 	taste_description = "the inside of a reactor"
 
+/datum/reagent/uranium/uraniumvirusfood/stable/on_mob_life(mob/living/carbon/M)
+	if(toxpwr)
+		M.adjustToxLoss(toxpwr*REM, 0)
+		. = TRUE
+	..()
+
```
Comment (Hopekz) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	M.adjustToxLoss(toxpwr*REM, 0)
	..()

```

**Review by Hopekz — CHANGES_REQUESTED** [review]
- useless if statements that cannot possibly be not true.
- You don't need to return true for these; it doesn't hurt but a weird practice and inconsistent with rest of codebase.

**Comment by boodaliboo** [comment]
thanks hopek

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 9]
```diff
@@ -1509,45 +1509,88 @@
 /datum/reagent/toxin/mutagen/mutagenvirusfood
 	name = "mutagenic agar"
 	color = "#A3C00F" // rgb: 163,192,15
+	var/toxpwr = 1.5
 	taste_description = "sourness"
 
+/datum/reagent/toxin/mutagen/mutagenvirusfood/on_mob_life(mob/living/carbon/M)
+	M.adjustToxLoss(toxpwr*REM, 0)
+	..()
+
```
Comment (Hopekz) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	taste_description = "sourness"

/datum/reagent/toxin/mutagen/mutagenvirusfood/on_mob_life(mob/living/carbon/M)
	M.adjustToxLoss(1.5*REM, 0)
	..()

```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 10]
```diff
@@ -1509,45 +1509,88 @@
...
 /datum/reagent/toxin/mutagen/mutagenvirusfood/sugar
 	name = "sucrose agar"
 	color = "#41B0C0" // rgb: 65,176,192
+	var/toxpwr = 0.5
 	taste_description = "sweetness"
+	
+/datum/reagent/toxin/mutagen/mutagenvirusfood/sugar/on_mob_life(mob/living/carbon/M)
+	M.adjustToxLoss(toxpwr*REM, 0)
+	..()
```
Comment (Hopekz) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	taste_description = "sweetness"
	
/datum/reagent/toxin/mutagen/mutagenvirusfood/sugar/on_mob_life(mob/living/carbon/M)
	M.adjustToxLoss(0.5*REM, 0)
	..()
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 11]
```diff
@@ -1509,45 +1509,88 @@
...
 
 /datum/reagent/medicine/synaptizine/synaptizinevirusfood
 	name = "virus rations"
 	color = "#D18AA5" // rgb: 209,138,165
+	var/toxpwr = 0.25
 	taste_description = "bitterness"
+	
+/datum/reagent/medicine/synaptizine/synaptizinevirusfood/on_mob_life(mob/living/carbon/M)
+	M.adjustToxLoss(toxpwr*REM, 0)
+	..()
```
Comment (Hopekz) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	taste_description = "bitterness"
	
/datum/reagent/medicine/synaptizine/synaptizinevirusfood/on_mob_life(mob/living/carbon/M)
	M.adjustToxLoss(0.25*REM, 0)
	..()
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 12]
```diff
@@ -1509,45 +1509,88 @@
...
 /datum/reagent/toxin/plasma/plasmavirusfood
 	name = "virus plasma"
 	color = "#A69DA9" // rgb: 166,157,169
+	var/toxpwr = 1
 	taste_description = "bitterness"
 	taste_mult = 1.5
+	
+/datum/reagent/toxin/plasma/plasmavirusfood/on_mob_life(mob/living/carbon/M)
+	M.adjustToxLoss(toxpwr*REM, 0)
+	..()
```
Comment (Hopekz) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	taste_description = "bitterness"
	taste_mult = 1.5
	
/datum/reagent/toxin/plasma/plasmavirusfood/on_mob_life(mob/living/carbon/M)
	M.adjustToxLoss(REM, 0)
	..()
```
Multiplying something by 1 is useless

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 13]
```diff
@@ -1509,45 +1509,88 @@
...
 /datum/reagent/toxin/plasma/plasmavirusfood/weak
 	name = "weakened virus plasma"
 	color = "#CEC3C6" // rgb: 206,195,198
+	var/toxpwr = 0.5
 	taste_description = "bitterness"
 	taste_mult = 1.5
 
+/datum/reagent/toxin/plasma/plasmavirusfood/weak/on_mob_life(mob/living/carbon/M)
+	M.adjustToxLoss(toxpwr*REM, 0)
+	..()
```
Comment (Hopekz) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	taste_description = "bitterness"
	taste_mult = 1.5

/datum/reagent/toxin/plasma/plasmavirusfood/weak/on_mob_life(mob/living/carbon/M)
	M.adjustToxLoss(0.5*REM, 0)
	..()
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 14]
```diff
@@ -1509,45 +1509,88 @@
...
+
 /datum/reagent/uranium/uraniumvirusfood
 	name = "decaying uranium gel"
 	color = "#67ADBA" // rgb: 103,173,186
+	var/toxpwr = 2.5
 	taste_description = "the inside of a reactor"
 
+/datum/reagent/uranium/uraniumvirusfood/on_mob_life(mob/living/carbon/M)
+	M.adjustToxLoss(toxpwr*REM, 0)
+	..()
```
Comment (Hopekz) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	taste_description = "the inside of a reactor"

/datum/reagent/uranium/uraniumvirusfood/on_mob_life(mob/living/carbon/M)
	M.adjustToxLoss(2.5*REM, 0)
	..()
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 15]
```diff
@@ -1509,45 +1509,88 @@
...
+
 /datum/reagent/uranium/uraniumvirusfood/unstable
 	name = "unstable uranium gel"
+	var/toxpwr = 2
 	color = "#2FF2CB" // rgb: 47,242,203
 	taste_description = "the inside of a reactor"
 
+/datum/reagent/uranium/uraniumvirusfood/unstable/on_mob_life(mob/living/carbon/M)
+	M.adjustToxLoss(toxpwr*REM, 0)
+	..()
```
Comment (Hopekz) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	color = "#2FF2CB" // rgb: 47,242,203
	taste_description = "the inside of a reactor"

/datum/reagent/uranium/uraniumvirusfood/unstable/on_mob_life(mob/living/carbon/M)
	M.adjustToxLoss(2*REM, 0)
	..()
```

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 16]
```diff
@@ -1509,45 +1509,88 @@
...
+
 /datum/reagent/uranium/uraniumvirusfood/stable
 	name = "stable uranium gel"
 	color = "#04506C" // rgb: 4,80,108
+	var/toxpwr = 1.5
 	taste_description = "the inside of a reactor"
 
+/datum/reagent/uranium/uraniumvirusfood/stable/on_mob_life(mob/living/carbon/M)
+	M.adjustToxLoss(toxpwr*REM, 0)
+	..()
```
Comment (Hopekz) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]:
```suggestion
	taste_description = "the inside of a reactor"

/datum/reagent/uranium/uraniumvirusfood/stable/on_mob_life(mob/living/carbon/M)
	M.adjustToxLoss(1.5*REM, 0)
	..()
```

**Review by Hopekz — CHANGES_REQUESTED** [review]
This is to make it compile
You keep defining the same variable over and over.

Might as well just put the values inside of the adjustToxLoss proc but you can do whatever other solution you make for this like making this an actual parent variable and not defining it every time instead.

**Review by Hopekz — APPROVED** [review]
Code checks out. Going to let this marinate for a bit to collect opinions

**Comment by Hopekz** [comment]
Let this marinate for 3 additional days.
Seems good from what I can tell I'll let us give this a try.

**Pull request merged**